include/deal.II/distributed/tria.h

```diff
@@ -1083,10 +1095,14 @@ namespace parallel
         std::vector<char> src_data_fixed;
         std::vector<char> dest_data_fixed;
 
-        // TODO: buffers for p4est variable size transfer
-        //       std::vector<std::vector<size_t>> cumulative_sizes_var_per_cell;
-        //       std::vector<size_t> src_sizes_var, dest_sizes_var;
-        //       std::vector<char> src_data_var, dest_data_var;
+        /**
+         * Consecutive buffers designed for the variable size transfer
+         * functions of p4est.
+         */
+        std::vector<int>  src_sizes_variable;
+        std::vector<int>  dest_sizes_variable;
+        std::vector<char> src_data_variable;
+        std::vector<char> dest_data_variable;
```

Comment:
Yes, unfortunately, they have to stay member variables. :(
We use these buffers for both data transfers via refinement and serialization. And since we want to unpack data at arbitrary locations in the code, we also have to keep them as members.